apps/agenda/agenda.js

```diff
@@ -69,16 +75,18 @@ function showEvent(ev) {
   //var lines = [];
   if (ev.title) lines = g.wrapString(ev.title, g.getWidth()-10);
   var titleCnt = lines.length;
-  var start = getDate(ev.timestamp);
-  var end = getDate((+ev.timestamp) + (+ev.durationInSeconds));
+  var start = getDate(ev.timestamp, ev.allDay);
+  // All day events end at the midnight boundary of the following day. Here, we
+  // subtract one second for all day events so the days display correctly.
+  var end = getDate((+ev.timestamp) + (+ev.durationInSeconds) - ev.allDay, ev.allDay);
```

Comment:
While this looks clever, requires the reader to check your comment to understand why you'd subtract what's basically a boolean value from a timestamp. I'd be a bit more verbose just for the sake of clarity, what do you think?
```suggestion
  // All day events end at the midnight boundary of the following day. Here, we
  // subtract one second for all day events so the days display correctly.
  const allDayOffset = ev.allDay ? 1 : 0;
  var end = getDate((+ev.timestamp) + (+ev.durationInSeconds) - allDayOffset, ev.allDay);
```

Reply:
Hilariously, that's exactly what I had wanted to do, but decided to be "clever" since it's an embedded device with minimal resources, etc (in the name of efficiency). Even told the wife how I already hated what I'd done :)

Every time I do something "clever", I regret it, so I'm glad we agree! lol

Reply:
To avoid colliding with the "offset" concept in `getDate()`, I chose a slightly different name than you did above. Let me know if this is alright with you.

Reply:
Ahahah nice one. As a side note, I'm not sure what the jit compiler would do but I'd bet a constant like this would have negligible effect.
The name you put is even clearer, go ahead with it, thanks